quickstart/201-iot-hub-with-device-provisioning-service/main.tf

```diff
@@ -0,0 +1,165 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  location = var.resource_group_location
+  name     = random_pet.rg_name.id
+}
+
+# Create storage account & container
+resource "random_string" "sa_name" {
+  length  = 12
+  special = false
+  upper   = false
+}
+
+resource "azurerm_storage_account" "sa" {
+  name                     = random_string.sa_name.id
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = azurerm_resource_group.rg.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_container" "my_terraform_container" {
+  name                  = "mycontainer"
+  storage_account_name  = azurerm_storage_account.sa.name
+  container_access_type = "private"
+}
+
+
+# Create an Event Hub & Authorization Rule
+resource "random_pet" "eventhub_namespace_name" {
+  prefix = var.eventhub_namespace_name_prefix
+}
+
+resource "azurerm_eventhub_namespace" "namespace" {
+  name                = random_pet.eventhub_namespace_name.id
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+  sku                 = "Basic"
+}
+
+resource "azurerm_eventhub" "my_terraform_eventhub" {
+  name                = "myEventHub"
+  resource_group_name = azurerm_resource_group.rg.name
+  namespace_name      = azurerm_eventhub_namespace.namespace.name
+  partition_count     = 2
+  message_retention   = 1
+}
+
+resource "azurerm_eventhub_authorization_rule" "my_terraform_authorization_rule" {
+  resource_group_name = azurerm_resource_group.rg.name
+  namespace_name      = azurerm_eventhub_namespace.namespace.name
+  eventhub_name       = azurerm_eventhub.my_terraform_eventhub.name
+  name                = "acctest"
+  send                = true
+}
+
+
+# Create an IoT Hub
+resource "random_pet" "iothub_name" {
+  prefix = var.iothub_name_prefix
+  length = 1
+}
+
+resource "azurerm_iothub" "iothub" {
+  name                = random_pet.iothub_name.id
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+
+  sku {
+    name     = "S1"
+    capacity = "1"
```

Comment:
According to the [provider's schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/iothub/iothub_resource.go#L119-L122), the `capacity`'s type is `TypeInt`. Terraform will [auto convert](https://www.terraform.io/language/expressions/types#type-conversion) `string` to `number`, but could we change `"1"` to `1` to make type more precisely please? Thanks.